### PR TITLE
Tlpso 448 - Update libraries for security alerts

### PR DESCRIPTION
**Pull request by dlhaines**
Update libraries to avoid security alerts from GitHub.
Improve error messages.
Update tests.
Fix issue with WAPI catching Exception.  Have it catch StandardError instead. (Catching Exception is incorrect and made debugging very difficult.)

There remains a testing problem with WebMock.  Those tests are currently being skipped.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,10 +8,15 @@ gem 'sinatra'
...
 gem 'rack'
-gem 'rack-protection'
+# update for security issue reported by github
+gem 'rack-protection', '~> 1.5.5'
+# update for security issue reported by github
+gem 'rubyzip', '~> 1.2.1'
 gem 'rest-client'
 gem 'link_header'
-gem 'net-ldap', '<= 0.12.1'
+#gem 'net-ldap', '<= 0.12.1'
```
Comment (zqian) [inline Gemfile]: Let's remove the old version line.

File: Gemfile [hunk 2]
```diff
@@ -8,10 +8,15 @@ gem 'sinatra'
 gem 'sinatra-contrib'
 gem 'slim'
 gem 'rack'
-gem 'rack-protection'
+# update for security issue reported by github
+gem 'rack-protection', '~> 1.5.5'
```
Comment (zqian) [inline Gemfile]: Do we need to put the 1.5.5 version here?

File: Rakefile [hunk 3]
```diff
@@ -70,7 +70,19 @@ namespace :test do
...
 
-## default unit tests
+  ## The "current" test template is available to allow a place to specify tests
+  ## of current interest during development.  The files specified are whatever
+  ## is convenient at the moment.
+  Rake::TestTask.new do |t|
+    t.libs << "test"
+    t.name = "current"
+    t.description = "current tests of interest"
+    #    t.test_files = FileList['**/test_WAPI.rb']
```
Comment (zqian) [inline Rakefile]: remove this line?

File: server/spec/test_mneme_api_response.rb [hunk 4]
```diff
@@ -72,39 +72,43 @@ def test_get_mneme_json_data_todolms
     refute_nil jsonA, "check that file contents are understood as json"
```
[the remark this answers is not in the document]
Comment (dlhaines) [reply server/spec/test_mneme_api_response.rb]: I suggest keeping it in order to minimize the changes.

File: Gemfile (hunk 1 above)
Comment (dlhaines) [reply Gemfile]: Done.

File: Gemfile (hunk 2 above)
Comment (dlhaines) [reply Gemfile]: That is the safest thing to do.  It minimizes the changes in dependencies.  If this was a big release with full testing then upping to the most recent version would be reasonable but it is better not do tempt fate.

File: Rakefile (hunk 3 above)
Comment (dlhaines) [reply Rakefile]: It's good to have a sample line.  I added some clarifications to the text in the Rakefile.

**Pull request merged**